pallets/connectors/src/lib.rs

```diff
@@ -486,11 +506,137 @@ pub mod pallet {
+			// Check that the destination is not the local domain
+			ensure!(
+				receiver.domain() != Domain::Centrifuge,
+				Error::<T>::InvalidDomain
+			);
+			ensure!(
+				!CurrencyIdOf::<T>::is_tranche_token(currency_id),
+				Error::<T>::InvalidTransferCurrency
+			);
+			let currency = Self::try_get_general_index(currency_id)?;
+
+			// FIXME: Ensure we never transfer tokens from DomainA to DomainB
+			//	 Potentially related to asset metadata update
+			//	 https://github.com/centrifuge/centrifuge-chain/pull/1393
+
+			// Transfer to the domain account for bookkeeping
+			T::Tokens::transfer(
+				currency_id,
+				&who,
+				&DomainLocator::<Domain> {
+					domain: receiver.domain(),
+				}
+				.into_account_truncating(),
+				amount,
+				false,
+			)?;
+
+			Self::do_send_message(
+				who.clone(),
+				Message::Transfer {
+					amount,
+					currency,
+					sender: who
+						.encode()
+						.try_into()
+						.map_err(|_| DispatchError::Other("Conversion to 32 bytes failed"))?,
+					receiver: receiver.address(),
+				},
+				receiver.domain(),
+			)?;
+
+			Ok(())
+		}
+
+		/// Add a currency to the set of known currencies on the domain derived
+		/// from the given currency.
+		#[pallet::weight(10_000 + T::DbWeight::get().writes(1).ref_time())]
+		#[pallet::call_index(8)]
+		pub fn add_currency(origin: OriginFor<T>, currency_id: CurrencyIdOf<T>) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			let currency = Self::try_get_general_index(currency_id)?;
+			// FIXME: Derive domain from currency
+			// locked by https://github.com/centrifuge/centrifuge-chain/pull/1393
+			// let domain_info = AssetRegistry::metadata(currency_id)?.location.into();
+			let domain = Domain::EVM(1284);
+
+			ensure!(domain != Domain::Centrifuge, Error::<T>::InvalidDomain);
+
+			Self::do_send_message(
+				who,
+				Message::AddCurrency {
+					currency,
+					// FIXME: Derive EVM address from currency
+					// Blocked by https://github.com/centrifuge/centrifuge-chain/pull/1393
+					evm_address: [0u8; 20],
+				},
+				domain,
+			)?;
+
+			Ok(())
+		}
+
+		/// Allow a currency to be used as a pool currency and to invest in a
+		/// pool on the domain derived from the given currency.
+		#[pallet::call_index(90)]
+		#[pallet::weight(10_000 + T::DbWeight::get().writes(1).ref_time())]
+		pub fn allow_pool_currency(
+			origin: OriginFor<T>,
+			pool_id: PoolIdOf<T>,
+			currency_id: CurrencyIdOf<T>,
+		) -> DispatchResult {
+			// TODO(subsequent PR): In the future, should be permissioned by trait which
+			// does not exist yet See spec: https://centrifuge.hackmd.io/SERpps-URlG4hkOyyS94-w?view#fn-update_member
+
+			// TODO(@review): According to spec, this should be restricted to
+			// `AdminOrigin`. However, `do_send_message` requires a 32-byte address for the
+			// payment of the fee. We could set the treasury.
+			// https://centrifuge.hackmd.io/SERpps-URlG4hkOyyS94-w?view#fn-add_pool_currency
```

Comment:
I think for now, I would just make this callable by anyone and only allow the equivalent currency on the target domain of the local pool currency. So if the pool currency is Statemint USDC, we would allow USDC on all other domains by default.

In the future we can figure out what would be the right flow to support multiple currencies, but I don't think we need to solve that now.

Reply:
See comment above.